application/cms/models.py

```diff
@@ -189,8 +197,10 @@ class DataSourceInMeasureVersion(db.Model):
     # metadata
     __tablename__ = "data_source_in_measure_version"
     __table_args__ = (
-        ForeignKeyConstraint(["data_source_id"], ["data_source.id"], name="data_source_in_page_data_source_id_fkey"),
-        ForeignKeyConstraint(["measure_version_id"], ["measure_version.id"]),
+        ForeignKeyConstraint(
+            ["data_source_id"], ["data_source.id"], name="data_source_in_page_data_source_id_fkey", ondelete="restrict"
```

Comment:
Are we going to end up with a bunch of orphaned data sources this way?  Maybe preserving data sources is premature, and we should just delete them along with the measure version for now?  And then if we ever rationalise data sources to be one-to-many we can re-jig the cascading then?

Reply:
That's not quite how these cascades work.

You don't seem to be able to cascade through join tables. So the two cascades on this join table mean the following:

When a measure version is deleted, cascade to the join table and delete any entries in the join table that reference the deleted measure version.

When a data source is deleted, cascade to the join table and - if there are any references to that data source - restrict (i.e. block) the delete. 

This means that you can delete a measure version and it will drop the entry from the join table (but the data source will not be dropped), but if you try to drop a data source that is referenced in the join table (i.e. by a measure version), it will fail.

Reply:
Aha OK that makes sense I think.  I suspect we already do drop data sources when measure versions are deleted, but it might not hurt to double check.  I don't necessarily love the join table, but removing it seems like more work than we need to do right now?  And maybe it will be useful in future.

Reply:
Yep, I think leaving the join table in now is probably best as well.